add function to rank prevalence by region

DanielleCornish opened the pull request:
add new column to dataset with ranks of prevalence from 1 (highest %
prevalence) to 13 (lowest)

DanielleCornish committed:
add function to rank prevalence by region

add new column to dataset with ranks of prevalence from 1 (highest %
prevalence) to 13 (lowest)

KathrynLittleboy commented on resources/r/model.R:
```diff
@@ -60,6 +60,13 @@ manipulate_regions_for_shapefile <- function(region_prevalence) {
+create_rank_variable_by_region <- function(thirteen_level_NHS_regional_prevalence){
+  thirteen_level_NHS_regional_prevalence$rank <- NA
+  thirteen_level_NHS_regional_prevalence$rank[order(-thirteen_level_NHS_regional_prevalence$prevalence)] <- 1:nrow(thirteen_level_NHS_regional_prevalence)
+  return(thirteen_level_NHS_regional_prevalence)
```
Can you add an extra line break before the 'return' line so it's consistent with the other functions?

DanielleCornish replied:
Done 😄

DanielleCornish committed:
tidy up layout

add line breaks for consistency

KathrynLittleboy commented on resources/r/model.R:
```diff
@@ -62,10 +62,11 @@ manipulate_regions_for_shapefile <- function(region_prevalence) {
   thirteen_level_NHS_regional_prevalence$rank[order(-thirteen_level_NHS_regional_prevalence$prevalence)] <- 1:nrow(thirteen_level_NHS_regional_prevalence)
-  return(thirteen_level_NHS_regional_prevalence)
+  
+  return(regional_prevalence_with_ranks)
```
This doesn't seem to be working now you've changed it from 'thirteen_level_nhs_regional_prevalence', 'regional_prevalence_with_ranks' isn't used anywhere in the function so it can't return it?

KathrynLittleboy commented on resources/r/model.R:
```diff
@@ -62,10 +62,11 @@ manipulate_regions_for_shapefile <- function(region_prevalence) {
   
 #Add rank column/variable to dataset
 
-create_rank_variable_by_region <- function(thirteen_level_NHS_regional_prevalence){
```
We need to update the flowchart so the functions/data names match - looks like the flowchart currently combines it with writing the narrative so we should split this out into a separate step in the flowchart

DanielleCornish committed:
update function as per comments

change function to use thirteen_level_NHS_regional_prevalence dataset

DanielleCornish committed:
update flowchart

split out ranking prevalence and creating narrative

KathrynLittleboy reviewed (commented):
I think you need to add a white box after the green one. Rename the green box so it matches the name of the function in the code you wrote (rank_prevalence_by_region) and then add a white box called 'regional_prevalence_with_ranks' as this is what you've named the new dataset you are outputting

KathrynLittleboy reviewed (commented):
Just for clarity, we might want to change part of the next function (joining shapefile to regional prevalence data) so that where it refers to the '13 level nhs regional prevalence' (line 73 and 76) it will refer to the 'regional prevalence with ranks' dataset which is most up to date. It currently works fine as it doesn't really matter what you name these bits within the function, and you've already updated the 'run' part - but it might make it easier to read.

DanielleCornish committed:
update flowchart

DanielleCornish committed:
update dataset names in shapefile join function

change to refer to 'regional prevalence with ranks' instead of 'thirteen
level NHS regional prevalence'

KathrynLittleboy reviewed (approved):
Looks good to me
